src/NimBLEUUID.cpp

```diff
@@ -318,4 +320,34 @@ std::string NimBLEUUID::toString() {
     return ble_uuid_to_str(&m_uuid.u, buf);
 } // toString
 
+
+NimBLEUUID & NimBLEUUID::operator=(const NimBLEUUID & other) {
+    m_uuid.u.type = other.m_uuid.u.type;
```

Comment:
This should be moved inside the `if(other.m_valueSet) {` block.

Reply:
You are right, but I think the copy constructor can be left out completely